docs/performance/inserts/tuning.txt

```diff
@@ -0,0 +1,172 @@
+.. _config_tuning:
+
+================================
+Configuration Tuning for Inserts
+================================
+
+This document outlines a number of hardware and software configuration changes
+you can make to tune your setup for inserts performance.
+
+.. rubric:: Table of Contents
+
+.. contents::
+   :local:
+
+Hardware
+========
+
+Solid-State Drives
+------------------
+
+Switch to *Solid-State Drives* (SSDs) if you can.
+
+SSDs are generally much faster than hard-disk drives, and usually offer the
+most cost-effective hardware upgrade for a CrateDB setup.
+
+Use the ``iostat`` tool to display disk I/O statistics, which should include
+output like this:
+
+.. code-block:: text
+
+    sh$ iostat
+    ...
+    avg-cpu:  %user   %nice %system %iowait  %steal   %idle
+               0.53    0.01    0.05    0.01    0.00   99.41
+    ...
+
+
+Here, the most interesting metric is ``%iowait``. This tells you how
+time the CPU is waiting for disk I/O. You can expect better insert performance
+from CrateDB relative to how low this value is.
+
+CPUs
+----
+
+If your queries are complex, the additional CPU overhead during the
+:ref:`analyzing phase <parsing_analyzing>` of insert query processing might
+result in a CPU performance bottleneck.
+
+If this becomes the case, and simplifying your queries isn't an option,
+upgrading your CPUs will be a cost-effective way to boost CrateDB performance.
+
+RAM Disks
+---------
+
+TODO are there any circumstances under which we would advise using a ram disk
+
+Software
+========
+
+Server
+------
+
+Replicas
+........
+
+Biggest influence: replicas (esp 0 vs . 1+)
+
+(Because primary write first and only after it writes the replicas)
+
+TODO: need details. what are we recommending here?
+
+Shards and Partitioning
+.......................
+
+Number of shards / partitioning
+
+TODO: need details. what are we recommending here?
+
+Table Schema
+............
+
+(Analyzer, INDEX OFF, Primary Key)
+
+TODO: need details. what are we recommending here?
+
+Translog
+........
+
+https://crate.io/docs/crate/reference/en/latest/sql/reference/create_table.html#translog-durability
+
+TODO: need details. what are we recommending here?
```

Comment:
Again, depends on the use-case/requirements. `ASYNC` leads to more performance, but bigger chance for data loss in case of disk failures. `REQUEST` is the default due to that.